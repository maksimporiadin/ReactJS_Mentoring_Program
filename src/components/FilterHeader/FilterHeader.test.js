import 'jsdom-global/register';
import React from 'react';

import { FilterHeader } from './index';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

describe('FilterHeader', () => {
    let wrapper;
    const defaultProps = {
        searchBy: 'title',
        inputValue: 'X-Men',
        onFilterChange() {},
        onChangeSearchBy() {},
        onSubmitRequest() {}
    };

    beforeEach(()=>{
        jest.resetAllMocks();
        wrapper = shallow(<FilterHeader {...defaultProps} />)
    });

    it('is renders', () => {
        expect(wrapper).toMatchSnapshot();
    });
});