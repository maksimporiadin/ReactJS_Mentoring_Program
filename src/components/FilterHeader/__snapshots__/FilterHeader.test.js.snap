// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterHeader is renders 1`] = `
<div
  className="filter"
>
  <form
    onSubmit={[Function]}
  >
    <h2
      className="filter-title"
    >
      Find your movie
    </h2>
    <Input
      className="form-control filter-input"
      inputConfig={
        Object {
          "placeholder": "Enter please search by title",
          "value": "X-Men",
        }
      }
      onChange={[Function]}
    />
    <SearchParam
      onChange={[Function]}
      searchBy="title"
    >
      <Button
        className="btn-primary btn-lg filter-button"
        name="SEARCH"
      />
    </SearchParam>
  </form>
</div>
`;
